File: Back End/Frontend/script/START.py
#!/usr/bin/env python3
# E-Kost Manager - Quick Start Script
# Python Version

import os
import sys
import subprocess
import webbrowser
import platform

def clear_screen():
    """Clear the screen based on the operating system"""
    os.system('cls' if os.name == 'nt' else 'clear')

def show_menu():
    """Display the main menu"""
    clear_screen()
    print("========================================")
    print("   E-Kost Manager - Quick Start")
    print("========================================")
    print()
    print("Pilih cara menjalankan aplikasi:")
    print()
    print("1. Standalone Mode (Tidak perlu install apapun)")
    print("2. Development Mode (Perlu Node.js)")
    print("3. Buka Dokumentasi")
    print("4. Keluar")
    print()

def start_standalone():
    """Start the application in standalone mode"""
    print()
    print("Membuka aplikasi dalam mode standalone...")
    
    standalone_path = os.path.join(os.getcwd(), 'standalone.html')
    
    if os.path.exists(standalone_path):
        try:
            webbrowser.open(standalone_path)
            print()
            print("Aplikasi dibuka di browser!")
            print("Jika browser tidak terbuka, buka file standalone.html manual")
        except Exception as e:
            print(f"Gagal membuka browser: {e}")
            print("Silakan buka file standalone.html manual")
    else:
        print("File standalone.html tidak ditemukan!")
    
    print()
    input("Tekan Enter untuk melanjutkan...")

def start_development():
    """Start the application in development mode"""
    print()
    print("Checking Node.js installation...")
    
    try:
        result = subprocess.run(['node', '--version'], 
                              capture_output=True, text=True, check=True)
        print(f"Node.js terinstall! Version: {result.stdout.strip()}")
    except (subprocess.CalledProcessError, FileNotFoundError):
        print("Node.js tidak terinstall!")
        print("Silakan install Node.js dari https://nodejs.org/")
        print("Atau gunakan opsi 1 (Standalone Mode)")
        print()
        input("Tekan Enter untuk melanjutkan...")
        return
    
    print()
    print("Installing dependencies...")
    
    try:
        subprocess.run(['npm', 'install'], check=True)
        print("Dependencies berhasil diinstall!")
    except subprocess.CalledProcessError:
        print("Gagal install dependencies!")
        print("Silakan gunakan opsi 1 (Standalone Mode)")
        print()
        input("Tekan Enter untuk melanjutkan...")
        return
    
    print()
    print("Starting development server...")
    print()
    print("Aplikasi akan berjalan di: http://localhost:3000")
    print()
    print("Tekan Ctrl+C untuk menghentikan server")
    print()
    
    try:
        subprocess.run(['npm', 'run', 'dev'])
    except KeyboardInterrupt:
        print("\nServer dihentikan.")
    except subprocess.CalledProcessError as e:
        print(f"Gagal menjalankan development server: {e}")
        input("Tekan Enter untuk melanjutkan...")

def open_docs():
    """Open documentation files"""
    print()
    print("Membuka dokumentasi...")
    
    docs = ['README.md', 'QUICK_START.md', 'START_HERE.md']
    
    for doc in docs:
        doc_path = os.path.join(os.getcwd(), doc)
        if os.path.exists(doc_path):
            try:
                webbrowser.open(doc_path)
            except Exception as e:
                print(f"Gagal membuka {doc}: {e}")
    
    print("Dokumentasi dibuka!")
    print()
    input("Tekan Enter untuk melanjutkan...")

def main():
    """Main function"""
    while True:
        show_menu()
        choice = input("Masukkan pilihan (1-4): ")
        
        if choice == '1':
            start_standalone()
        elif choice == '2':
            start_development()
        elif choice == '3':
            open_docs()
        elif choice == '4':
            print()
            print("Terima kasih!")
            print()
            break
        else:
            print("Pilihan tidak valid!")
            input("Tekan Enter untuk melanjutkan...")

if __name__ == "__main__":
    main()

File: Front End/script/START.js
// E-Kost Manager - Quick Start Script
// Node.js Version

const { exec, spawn } = require('child_process');
const fs = require('fs');
const path = require('path');

function showMenu() {
    console.clear();
    console.log('========================================');
    console.log('   E-Kost Manager - Quick Start');
    console.log('========================================');
    console.log('');
    console.log('Pilih cara menjalankan aplikasi:');
    console.log('');
    console.log('1. Standalone Mode (Tidak perlu install apapun)');
    console.log('2. Development Mode (Perlu Node.js)');
    console.log('3. Buka Dokumentasi');
    console.log('4. Keluar');
    console.log('');
}

function startStandalone() {
    console.log('');
    console.log('Membuka aplikasi dalam mode standalone...');
    
    const standalonePath = path.join(__dirname, 'standalone.html');
    
    if (fs.existsSync(standalonePath)) {
        const platform = process.platform;
        let command;
        
        switch (platform) {
            case 'win32':
                command = 'start';
                break;
            case 'darwin':
                command = 'open';
                break;
            default:
                command = 'xdg-open';
        }
        
        exec(`${command} "${standalonePath}"`, (error) => {
            if (error) {
                console.log('Gagal membuka browser. Silakan buka file standalone.html manual');
            } else {
                console.log('Aplikasi dibuka di browser!');
                console.log('Jika browser tidak terbuka, buka file standalone.html manual');
            }
        });
    } else {
        console.log('File standalone.html tidak ditemukan!');
    }
    
    console.log('');
    console.log('Tekan Enter untuk melanjutkan...');
    process.stdin.read();
}

function startDevelopment() {
    console.log('');
    console.log('Checking Node.js installation...');
    
    exec('node --version', (error, stdout, stderr) => {
        if (error) {
            console.log('Node.js tidak terinstall!');
            console.log('Silakan install Node.js dari https://nodejs.org/');
            console.log('Atau gunakan opsi 1 (Standalone Mode)');
            console.log('');
            console.log('Tekan Enter untuk melanjutkan...');
            process.stdin.read();
            return;
        }
        
        console.log(`Node.js terinstall! Version: ${stdout.trim()}`);
        console.log('');
        console.log('Installing dependencies...');
        
        exec('npm install', (error, stdout, stderr) => {
            if (error) {
                console.log('Gagal install dependencies!');
                console.log('Silakan gunakan opsi 1 (Standalone Mode)');
                console.log('');
                console.log('Tekan Enter untuk melanjutkan...');
                process.stdin.read();
                return;
            }
            
            console.log('Dependencies berhasil diinstall!');
            console.log('');
            console.log('Starting development server...');
            console.log('');
            console.log('Aplikasi akan berjalan di: http://localhost:3000');
            console.log('');
            console.log('Tekan Ctrl+C untuk menghentikan server');
            console.log('');
            
            const devProcess = spawn('npm', ['run', 'dev'], { stdio: 'inherit' });
            
            devProcess.on('error', (error) => {
                console.log('Gagal menjalankan development server:', error.message);
                console.log('Tekan Enter untuk melanjutkan...');
                process.stdin.read();
            });
        });
    });
}

function openDocs() {
    console.log('');
    console.log('Membuka dokumentasi...');
    
    const docs = ['README.md', 'QUICK_START.md', 'START_HERE.md'];
    const platform = process.platform;
    let command;
    
    switch (platform) {
        case 'win32':
            command = 'start';
            break;
        case 'darwin':
            command = 'open';
            break;
        default:
            command = 'xdg-open';
    }
    
    docs.forEach(doc => {
        const docPath = path.join(__dirname, doc);
        if (fs.existsSync(docPath)) {
            exec(`${command} "${docPath}"`, (error) => {
                if (error) {
                    console.log(`Gagal membuka ${doc}`);
                }
            });
        }
    });
    
    console.log('Dokumentasi dibuka!');
    console.log('');
    console.log('Tekan Enter untuk melanjutkan...');
    process.stdin.read();
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    function askChoice() {
        showMenu();
        rl.question('Masukkan pilihan (1-4): ', (choice) => {
            switch (choice) {
                case '1':
                    startStandalone();
                    break;
                case '2':
                    startDevelopment();
                    break;
                case '3':
                    openDocs();
                    break;
                case '4':
                    console.log('');
                    console.log('Terima kasih!');
                    console.log('');
                    rl.close();
                    return;
                default:
                    console.log('Pilihan tidak valid!');
                    console.log('Tekan Enter untuk melanjutkan...');
                    process.stdin.read();
                    askChoice();
            }
        });
    }
    
    askChoice();
}

main();